### PR TITLE
Handle "Color color" from VB completion

### LINKED ISSUE #3086: VB intellisense doesn't show "color color" cases
As per the screenshot below, VB completion list isn't showing static members off of an instance, even though the "color color" rule lets the compiler happily consume them.

intellisense-vb-color-color

WHAT I EXPECT: completion list should show Y.
WHAT I GET: it doesn't.

This repro shows a const member "Y". The same bug also manifests for *shared* members in class C.

### PR DESCRIPTION
**Pull request by rchande**
Fixes #3086.

Please review: @balajikris @brettfo @dpoeschl @jasonmalinowski @Pilchie

### CODE REVIEW

**Comment by rchande** [comment]
Note that I didn't add a test for case (1) mentioned by @ljw1004 because there's already a test covering it.

File: src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb [hunk 1]
```diff
@@ -250,16 +250,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
                         End If
                 End Select
 
-                If inNameOfExpression Then
+                ' Check for color color
+                Dim speculativeTypeBinding = context.SemanticModel.GetSpeculativeTypeInfo(context.Position, leftExpression, SpeculativeBindingOption.BindAsTypeOrNamespace)
+                If firstSymbol.GetSymbolType() Is speculativeTypeBinding.Type Then
```
Comment (Pilchie) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb]: Does this handle aliases right?

What happens with something like:

```VB
Imports Str = System.String

Module Program
    Sub Main(args As String())
        Str.$$
    End Sub

    Dim Str As Str
End Module
```

**Comment by rchande** [comment]
@Pilchie Good point, will fix that.

**Comment by rchande** [comment]
Actually, @Pilchie, aliases work just fine. I had already built a different branch when I checked.

**Comment by Pilchie** [comment]
Define "work fine". Do they show `Shared` members in that case or not?  My quick testing showed that the compiler does *NOT* allow the access if it's an alias.

**Comment by rchande** [comment]
@Pilchie Added more commits, we won't show statics off aliases.

**Comment by rchande** [comment]
Will squash before merging.

**Comment by Pilchie** [comment]
:+1:

**Comment by dpoeschl** [comment]
LGTM

**Comment by MattGertz** [comment]
Approved 6/1 by ML Shiproom.  This meets the regression bar and is very low risk.

**Comment by rchande** [comment]
test this please

**Comment by dpoeschl** [comment]
retest this please

**Pull request merged**